app/src/main/java/com/allan/boardbuddies/fragments/TextDialogFragment.java

```diff
@@ -0,0 +1,72 @@
+package com.allan.boardbuddies.fragments;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import android.view.WindowManager;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+import com.allan.boardbuddies.R;
+
+public class TextDialogFragment extends DialogFragment {
+
+    private EditText editText;
+
+    public static TextDialogFragment newInstance(String in) {
+        TextDialogFragment f = new TextDialogFragment();
+
+        Bundle args = new Bundle();
+        args.putString("text", in);
+        f.setArguments(args);
+
+        return f;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setStyle(DialogFragment.STYLE_NO_FRAME, android.R.style.Theme_Holo);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.parseColor("#4D000000")));
+        return inflater.inflate(R.layout.text_dialog, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
```

Comment:
nit: I think generally what i've seen is that we want to contain the manipulation of soft input mode to within the fragment (so that we don't mess with other fragments, more relevant when you have more going on in the app). So usually we would set this in onResume and then set it back to ADJUST_NOTHING in onPause

Reply:
idgi... Is it not contained to this fragment rn?

Reply:
I see, you're setting this on the dialog's window. In that case it probably doesn't matter.

What I meant was, say you set the soft input mode for the activity's window, then maybe on some other fragment it will also use the same soft input mode. It's good practice to reset it so we know exactly what to expect.